src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -733,6 +733,38 @@ namespace winrt::TerminalApp::implementation
         {
             _noMatchesText().Visibility(Visibility::Collapsed);
         }
+
+        if (_currentMode == CommandPaletteMode::CommandlineMode)
+        {
+            ParsedCommandLineText(L"");
+
+            const auto commandLine = _getTrimmedInput();
+            if (!commandLine.empty())
+            {
+                ExecuteCommandlineArgs args{ commandLine };
+                ::TerminalApp::AppCommandlineArgs appArgs;
```

Comment:
Part of me wonders if the CommandPalette should just keep a single `AppCommandlineArgs` instance as a member, and then reset it and re-use it for parsing each time. 

That being said, this solution certainly _does work_

Reply:
@zadjii-msft - I see your point. I allowed this to myself, since today we do allocate an instance upon every execution. However, now we will actually allocate an instance on every character in the search - so it makes sense to cache it. I will see if there is a good way to reset the instance - since the internal reset we have there now is partial IIRC.